client/rpc/src/eth/fee.rs

```diff
@@ -71,12 +71,16 @@ where
 			self.backend.as_ref(),
 			Some(newest_block),
 		) {
-			let header = match self.client.header(id) {
+			let Ok(hash) = self.client.expect_block_hash_from_id(&id) else {
+				return Err(internal_err(format!("Failed to retrieve block hash at {id}")));
+			};
+			let header = match self.client.header(hash) {
 				Ok(Some(h)) => h,
 				_ => {
 					return Err(internal_err(format!("Failed to retrieve header at {}", id)));
 				}
 			};
+			// Canonical chain block number.
 			let number = match self.client.number(header.hash()) {
```

Comment:
Not sure if `expect_block_number_from_id` will be enough here, added extra canonicalization just to be safe.

Reply:
I think it should be safe to use `expect_block_number_from_id` since as per [docs](https://paritytech.github.io/substrate/master/src/sp_blockchain/backend.rs.html#58) it calls the same method `HeaderBackend::number(...)`  to retrieve the block number. This will also simplify the code.